haskell.nix/library/.github/workflows/check-outdated.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  schedule:
+    - cron: '0 10 * * 1'
+
+jobs:
+  check-outdated:
+    runs-on: [self-hosted, nix]
+    outputs:
+      msg: ${{ steps.outdated.outputs.outdated }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: check outdated packages
+        id: outdated
+        run: |
+          function output {
+            echo "outdated<<EOF" >> $GITHUB_OUTPUT
+            echo "Repository https://github.com/${{ github.repository }} has outdated haskell dependencies" >> $GITHUB_OUTPUT
+            echo "$msg" >> $GITHUB_OUTPUT
+            echo "EOF" >> $GITHUB_OUTPUT
+          }
+
+          nix develop -c cabal update
+          trap 'output' ERR
+          msg=$(nix develop -c cabal outdated --exit-code)
+
+  report-outdated:
+    runs-on: [self-hosted, nix]
+    if: failure()
+    needs: check-outdated
+    steps:
+      - name: Report outdated haskell dependencies
+        env:
+          MSG: ${{needs.check-outdated.outputs.msg}}
+        run: |
+          curl -XPOST https://slack.com/api/chat.postMessage \
+            -H "Authorization: Bearer ${{ secrets.SLACK_TOKEN }}" \
+            -d "channel=libraries" \
```

Comment:
Also I would mention somewhere that if the user of this template wants this workflow to send messages to a different channel, they can change this line. We have dedicated channels for some libraries, e. g. `universum` and `o-clock`.